src/commands/tsgen.ts

```diff
@@ -91,38 +104,61 @@ export default class TypeScriptCodeGeneratorCommand extends Command {
         this.error('Please provide an output path.', {exit: 2})
       }
 
+      let region = ''
+      switch (this.region.name) {
+      case 'NA':
+        region = 'US'
+        break
+      case 'AZURE-NA':
+        region = 'AZURE_NA'
+        break
+      case 'AZURE-EU':
+        region = 'AZURE_EU'
+        break
+      case 'GCP-NA':
+        region = 'GCP_NA'
+        break
+      default:
+        region = this.region.name.toUpperCase()
+      }
+
       const config: StackConnectionConfig = {
         apiKey: token.apiKey,
         token: token.token,
-        region: (this.region.name === 'NA') ? 'us' : this.region.name.toLowerCase(),
+        region: region,
         environment: token.environment || '',
-        branch: branch || null,
+        branch: branch || undefined,
       }
 
+      // Generate the GraphQL schema TypeScript definitions
       if (flags['api-type'] === 'graphql') {
-        const result = await generateGraphQLTypeDef(config, outputPath, namespace)
-        if (result) {
-          this.log(`Successfully added the GraphQL schema type definitions to '${result.outputPath}'.`)
-        } else {
-          this.log('No schema found in the stack! Please use a valid stack.')
+        try {
+          const result = await graphqlTS({...config, namespace: namespace})
+
+          fs.writeFileSync(outputPath, result)
+          this.log(`Successfully added the GraphQL schema type definitions to '${outputPath}'.`)
+        } catch (error: any) {
+          this.error(error.error_message, {exit: 1})
```

Comment:
@praveen-mohan-cs , @meetstack19 Are you certain that the error_message key contains all of the error messages we receive? My understanding is that it may also throw a message in the message or errors property.

Reply:
Hi @aman19K ,

We’ve implemented the logic to fetch error messages from the GraphQL API Response and display them appropriately. Additionally, we’ve added handling for scenarios where the error_message key is not present.

You can check out the implementation here: [Link](https://github.com/contentstack/types-generator/blob/3e294b8667a6d6565a2bc46c617faf375d2a30fa/src/graphqlTS/index.ts#L69C1-L88C4)

CC:- @praveen-mohan-cs @PriteshJain